integration/local-gadget/integration_test.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"math/rand"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	containercollection "github.com/kinvolk/inspektor-gadget/pkg/container-collection"
+)
+
+const (
+	ContainerRuntimeDocker     = "docker"
+	ContainerRuntimeContainerd = "containerd"
+	ContainerRuntimeCRIO       = "cri-o"
+)
+
+var supportedContainerRuntimes = []string{ContainerRuntimeDocker, ContainerRuntimeContainerd, ContainerRuntimeCRIO}
+
+var (
+	integration      = flag.Bool("integration", false, "run integration tests")
+	containerRuntime = flag.String("container-runtime", "", "allows to skip tests that are not supported on a given container-runtime")
+)
+
+func testMain(m *testing.M) int {
+	flag.Parse()
+	if !*integration {
+		fmt.Println("Skipping integration test.")
+		return 0
+	}
+
+	if *containerRuntime != "" {
+		found := false
+		for _, val := range supportedContainerRuntimes {
+			if *containerRuntime == val {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			fmt.Fprintf(os.Stderr, "Error: invalid argument '-container-runtime': %q. Valid values: %s\n",
+				*containerRuntime, strings.Join(supportedContainerRuntimes, ", "))
+			return -1
+		}
+	}
+
+	seed := time.Now().UTC().UnixNano()
+	rand.Seed(seed)
+	fmt.Printf("using random seed: %d\n", seed)
+
+	fmt.Println("Start running tests:")
+	return m.Run()
+}
+
+func TestMain(m *testing.M) {
+	os.Exit(testMain(m))
+}
+
+func TestListContainers(t *testing.T) {
+	ctx := context.Background()
+	prefix := "list-containers"
+	podID, err := runTestContainer(ctx, *containerRuntime, prefix, "docker.io/library/alpine", "sleep inf")
+	if err != nil {
+		t.Fatalf("failed to run container: %v", err)
+	}
+	defer removeTestContainer(ctx, *containerRuntime, podID)
+
+	out, err := run("local-gadget", "list-containers", "-o", "json", "--runtimes", *containerRuntime)
```

Comment:
I propose adding another test here using the `--containername` flag and then using a helper function that verifies there is only one result, and it matches the expected one. While the existing test (without the filtering) expects multiple results but only one must match the expected one.